### PR TITLE
Creation of base class for amp-story layers.

### LINKED ISSUE #13710: Add a base class for amp-story layers
We currently have `AmpStoryGridLayer` and `AmpStoryCtaLayer`, which share no implementation.  We should add a common base class `AmpStoryLayer` that applies a CSS class name like `i-amphtml-story-layer` so that other components that need to get all layers have a way of doing so.

One concrete example of this is the page scaling service; it should get all layers and (maybe) apply a scaling factor to them; currently it's tightly coupled to `amp-story-grid-layer`, which we should avoid doing unless explicitly desired.

### PR DESCRIPTION
**Pull request by Enriqe**
Fixes #13710

Addition of a new base class for amp-story layers. Right now there are two layers `AmpStoryCtaLayer` and `AmpStoryGridLayer`.

This will make them share an implementation so that other components have a way to get all layers if they need to do so.

### CODE REVIEW

File: extensions/amp-story/0.1/amp-story-base-layer.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Base layer from which other layers in a story page extend from.
+ */
+
+import {Layout} from '../../../src/layout';
+
+export class AmpBaseLayer extends AMP.BaseElement {
```
Comment (newmuis) [inline extensions/amp-story/0.1/amp-story-base-layer.js]: In `buildCallback()` we can automatically add the CSS class name you created to the layer, with:

```javascript
this.element.classList.add('i-amphtml-story-layer');
```

And you'll need to make sure that `AmpStoryGridLayer` calls `super.buildCallback()` in its `buildCallback()` function.

File: extensions/amp-story/0.1/amp-story-base-layer.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
```
Comment (newmuis) [inline extensions/amp-story/0.1/amp-story-base-layer.js]: nit: this should be named `AmpStoryBaseLayer`

File: extensions/amp-story/0.1/amp-story.css [hunk 3]
```diff
@@ -268,24 +268,23 @@ amp-story:not([desktop]) >
   z-index: 2 !important;
 }
 
+.i-amphtml-story-layer, amp-story-cta-layer, amp-story-grid-layer {
```
Comment (newmuis) [inline extensions/amp-story/0.1/amp-story.css]: Actually, the more I look at this, the more I think I'm in favor of not including `.i-amphtml-story-layer` in this selector.  It's by *coincidence* that all `.i-amphtml-story-layer` implementations have these styles in common, not by *definition*, so I think I'd be more in favor of applying it to the layer implementations themselves (which, by definition, have these styles).

Comment (Enriqe) [reply extensions/amp-story/0.1/amp-story.css]: Makes sense 👍

**Pull request merged**